### PR TITLE
Bump psalm to 3.8.2

### PR DESCRIPTION
**Pull request by Lctrs**

### CODE REVIEW

**Comment by Lctrs** [comment]
It will need a good triage between false positives/real errors.

**Comment by Lctrs** [comment]
@Ocramius Some advices on this one would be really appreciated :)

**Comment by Ocramius** [comment]
I think I'm OK with this as-is for now, will review the baseline anyway though

File: psalm-baseline.xml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    <InvalidScalarArgument occurrences="2"/>
+  </file>
+  <file src="src/SourceLocator/Type/DirectoriesSourceLocator.php">
+    <DocblockTypeContradiction occurrences="1">
+      <code>is_string($directory)</code>
+    </DocblockTypeContradiction>
+  </file>
+  <file src="src/SourceLocator/Type/FileIteratorSourceLocator.php">
+    <DocblockTypeContradiction occurrences="1">
+      <code>$fileInfo instanceof SplFileInfo</code>
```
Comment (Ocramius) [inline psalm-baseline.xml]: `@param Iterator|SplFileInfo[] $fileInfoIterator` is wrong here: this is an `@param Iterator<SplFileInfo> $fileInfoIterator` instead

**Comment by Ocramius** [comment]
We should probably open a new issue for the comments above. Reaching 100% type coverage in psalm will be damn hard, but worth the effort.

If you can rebase this one, and open a new issue, we should be good to go here for now

**Comment by Lctrs** [comment]
@Ocramius Done :)

**Pull request merged**